src/cpu/testers/dr_trace_player/trace_player.cc

```diff
@@ -230,9 +265,35 @@ DRTracePlayer::getPacket(DRTraceReader::TraceRef &mem_ref)
         }
     }
 
-    return pkt;
-}
+    PacketPtr split_pkt = nullptr;
+
+    // In case of split packets when we want to send two requests.
+    // For the second request, the starting address
+    // will be split_addr and the size will be cacheLineSize - size
 
+    if (split_req) {
+
+        // Create the split request
+        RequestPtr split_req = std::make_shared<Request>(split_addr,
+                                    cacheLineSize - size, flags, requestorId);
+        split_req->setPC(curPC);
+
+        // Embed it in a packet
+        split_pkt = new Packet(split_req, cmd);
+
+        if (params().send_data) {
+            uint8_t* split_pkt_data = new uint8_t[split_req->getSize()];
+            split_pkt->dataDynamic(split_pkt_data);
+
+            if (cmd.isWrite()) {
+                std::fill_n(split_pkt_data, split_req->getSize(),
+                                            (uint8_t)requestorId);
+            }
+        }
+    }
+
+    return std::make_pair(pkt, split_pkt);
```

Comment:
I think the following works in modern C++.

```suggestion
    return {pkt, split_pkt};
```

Reply:
done